Do you see how to resolve this? Only happens with the gcc-7.4.0, see travis report.
```
/home/travis/build/ComputationalRadiationPhysics/alpaka/test/unit/atomic/src/AtomicTest.cpp: In function 'testAtomicCas<alpaka::acc::AccCpuSerial<std::integral_constant<unsigned long, 1ul>, unsigned long>, long long>(alpaka::acc::AccCpuSerial<std::integral_constant<unsigned long, 1ul>, unsigned long> const&, bool*, long long)void':
/home/travis/build/ComputationalRadiationPhysics/alpaka/include/alpaka/atomic/Op.hpp:246:45: error: assuming signed overflow does not occur when simplifying conditional to constant [-Werror=strict-overflow]
                     ref = ((old == compare) ? value : old);
```
Code is:
```c++
ALPAKA_FN_HOST_ACC auto operator()(        
    T * addr,                              
    T const & compare,                     
    T const & value) const                 
-> T                                       
{                                          
    auto const old(*addr);                 
    auto & ref(*addr);                     
    ref = ((old == compare) ? value : old);
    return old;                            
}                                          
```

--- a/include/alpaka/atomic/Op.hpp
+++ b/include/alpaka/atomic/Op.hpp
@@ -243,7 +243,17 @@ namespace alpaka
                 {
                     auto const old(*addr);
                     auto & ref(*addr);
+
+// gcc-7.4.0 assumes for an optimization that a signed overflow does not occur here.
+// That's fine, so ignore that warning.
+#if BOOST_COMP_GNUC && (BOOST_COMP_GNUC == BOOST_VERSION_NUMBER(7, 4, 0))
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wstrict-overflow"
+#endif
                     ref = ((old == compare) ? value : old);
+#if BOOST_COMP_GNUC && (BOOST_COMP_GNUC == BOOST_VERSION_NUMBER(7, 4, 0))
+#pragma GCC diagnostic pop
+#endif
                     return old;
                 }
             };

--- a/test/unit/atomic/src/AtomicTest.cpp
+++ b/test/unit/atomic/src/AtomicTest.cpp
@@ -950,21 +950,19 @@ struct TestTemplate
     template< typename TAcc >
     void operator()()
     {
-        // This test exceeds the maximum compilation time on MSVC.
-#if !defined(ALPAKA_CI)
         TestAtomicOperations<TAcc, unsigned char>::testAtomicOperations();
         TestAtomicOperations<TAcc, char>::testAtomicOperations();
         TestAtomicOperations<TAcc, unsigned short>::testAtomicOperations();
         TestAtomicOperations<TAcc, short>::testAtomicOperations();
-#endif
+
         TestAtomicOperations<TAcc, unsigned int>::testAtomicOperations();
         TestAtomicOperations<TAcc, int>::testAtomicOperations();
-#if !(defined(ALPAKA_CI) && BOOST_COMP_MSVC)
+
         TestAtomicOperations<TAcc, unsigned long>::testAtomicOperations();
         TestAtomicOperations<TAcc, long>::testAtomicOperations();
         TestAtomicOperations<TAcc, unsigned long long>::testAtomicOperations();
         TestAtomicOperations<TAcc, long long>::testAtomicOperations();
-#endif
+
         // Not all atomic operations are possible with floating point values.
         //TestAtomicOperations<TAcc, float>::testAtomicOperations();
         //TestAtomicOperations<TAcc, double>::testAtomicOperations();

--- a/test/unit/atomic/src/AtomicTest.cpp
+++ b/test/unit/atomic/src/AtomicTest.cpp
@@ -973,5 +973,9 @@ struct TestTemplate
 
 TEST_CASE( "atomicOperationsWorking", "[atomic]")
 {
-    alpaka::meta::forEachType< alpaka::test::acc::TestAccs >( TestTemplate() );
+    using TestAccs = alpaka::test::acc::EnabledAccs<
+        alpaka::dim::DimInt<1u>,
+        std::size_t>;
+
+    alpaka::meta::forEachType< TestAccs >( TestTemplate() );
 }

